### PR TITLE
ref(replay): Rewrite useReplayData to leverage useApiQuery

### LINKED ISSUE #61359: [Replay Details] Load error pages in parallel

### PR DESCRIPTION
**Pull request by ryan953**
Per the title, this is a rewrite of `useReplayData` that leverages `useApiQuery` and friends to cache data, which will make navigations back/forward much faster.

Also, this abstracts out the helper `useFetchReplayPaginatedData` so we can load both attachments (breadcrumbs and rrweb events) in parallel (as before), and now we're using it to load errors in parallel too (based on the count_errors field).

Fixes https://github.com/getsentry/sentry/issues/61359

### CODE REVIEW

File: static/app/utils/replays/hooks/useReplayData.tsx [hunk 1]
```diff
@@ -58,242 +45,140 @@ interface Result {
...
 
-const INITIAL_STATE: State = Object.freeze({
-  fetchError: undefined,
-  fetchingAttachments: true,
-  fetchingErrors: true,
-  fetchingReplay: true,
-});
-
-/**
- * A react hook to load core replay data over the network.
```
[the remark this answers is not in the document]
Comment (ryan953) [reply static/app/utils/replays/hooks/useReplayData.tsx:69]: the file was more of a rewrite, but yeah i can bring this back.

File: static/app/utils/api/useFetchParallelPages.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+interface State<Data> {
+  error: unknown;
+  getLastResponseHeader: ((header: string) => string | null) | undefined;
+  isError: boolean;
+  isFetching: boolean;
+  pages: Data[];
+}
+
+export default function useFetchParallelPages<Data>({
```
[the remark this answers is not in the document]
Comment (ryan953) [reply static/app/utils/api/useFetchParallelPages.tsx:49]: Can write those comments yeah!

I'm not sure exactly the utility of each of these things in other parts of the app. I expect if someone needs them they'll either write some convoluted thing themselves (which is what replays had before) or they can re-use some code. There are examples of code re-use across different 'product' folders inside `components/`, so hiding it in there won't be enough to hide the pattern from people completely either :p

[the remark this answers is not in the document]
Comment (ryan953) [reply static/app/utils/api/useFetchParallelPages.tsx:89]: wrote some stuff!

File: static/app/utils/api/useFetchParallelPages.tsx [hunk 3]
```diff
@@ -46,6 +46,46 @@ interface State<Data> {
...
 }
 
+/**
+ * A query hook to fetch a fixed number of list pages all at once.
+ *
+ * See also: `useFetchSequentialPages()`
+ *
+ * <WARNING>
+ *   Using this hook might not be a good idea!
+ *   Pagination is a good stratergy to limit the amount of data that a server
```
Comment (malwilley) [inline static/app/utils/api/useFetchParallelPages.tsx:56]: > stratergy

🙃

**Review by malwilley — APPROVED** [review]
Very helpful doc comments! Especially like the thorough warnings.

**Comment by ryan953** [comment]
I'm taking this in parts: https://github.com/getsentry/sentry/pull/63455 is the first.

**Pull request closed without merging**